pkg/controller/handlers/threads/threads.go

```diff
@@ -601,3 +601,37 @@ func (t *Handler) CopyTasksFromParent(req router.Request, _ router.Response) err
 
 	return nil
 }
+
+func MigrateOauthAppsFromParent(ctx context.Context, c kclient.Client) error {
```

Comment:
@thedadams I did change so that we use thread's oauthapp by default and in order to not break existing project, we have to migrate the oauthapp from agent to project.